src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -151,31 +147,105 @@ public void refreshMapForOrgId(String orgId) {
 		FhirClientProvider fhirClientProvider = new FhirClientProviderImpl((GenericClient) FhirClientAuthenticatorService.getFhirClient());
 		mapOfOrgHierarchy.put(orgId, ReportGeneratorFactory.INSTANCE.reportGenerator().getOrganizationHierarchy(fhirClientProvider, orgId));
 	}
-	
-	
-	
+
+	private String getKeycloakGroupId(String uid) {
+		RealmResource realmResource = FhirClientAuthenticatorService.getKeycloak().realm(appProperties.getKeycloak_Client_Realm());
+		try {
+			if (realmResource.groups().groups(uid, 0, Integer.MAX_VALUE, false).size() == 0){
+				return null;
+			}
+			return realmResource.groups().groups(uid, 0, Integer.MAX_VALUE, false).get(0).getId();
```

Comment:
`realmResource.groups().groups(uid, 0, Integer.MAX_VALUE, false)` The same code part is used twice. Extract it to a variable.